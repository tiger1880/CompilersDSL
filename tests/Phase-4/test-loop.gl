$ while 
$ for with optional arguments