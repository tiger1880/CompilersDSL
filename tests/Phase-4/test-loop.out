#include <iostream>
#include <vector>
#include <GL/glut.h>
#include <string>
#include <cstdlib>
#include <cmath>
#include <deque>
#include "standard_lib.hpp"

int a = 5;
int sum = 6;

int main(int argc, char **argv)
{

    for (int i = 0; i < a; i++)
    {
        sum += i;
    }

    while (sum > a)
    {
        sum -= 4;
        string s = "Helloooo!!!";
    }

    for (int j = 7; j > 8;)
    {
        j++;
    }

    for (int c = 5;;)
    {
        c--;
    }

    glutInit(&argc, argv);
    glutInitWindowSize(640, 480);
    glutInitWindowPosition(50, 50);
    glutCreateWindow("Figure");
    glutDisplayFunc(display);
    glutReshapeFunc(reshape);
    init();
    glutMainLoop();

    return 0;
}
